src/Compilers/CSharp/Test/Emit/PDB/TypeDefinitionDocumentTests.cs

```diff
@@ -0,0 +1,446 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using System.Text;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Debugging;
+using Microsoft.CodeAnalysis.Emit;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.PDB
+{
+    public class TypeDefinitionDocumentTests : CSharpTestBase
+    {
+        [Fact]
+        public void ClassWithMethod()
+        {
+            string source = @"
+class M
+{
+    public static void A()
+    {
+        System.Console.WriteLine();
+    }
+}
+";
+            TestTypeDefinitionDocuments(new[] { source });
+        }
+
+        [Fact]
+        public void NestedClassWithMethod()
+        {
+            string source = @"
+class C
+{
+    class N
+    {
+        public static void A()
+        {
+            System.Console.WriteLine();
+        }
+    }
+}
+";
+            TestTypeDefinitionDocuments(new[] { source },
+                ("C", "1.cs"));
+        }
+
+        [Fact]
+        public void EmptyClass()
+        {
+            string source = @"
+class O
+{
+}
+";
+            TestTypeDefinitionDocuments(new[] { source },
+                ("O", "1.cs"));
+        }
+
+
+        [Fact]
+        public void EmptyNestedClass()
+        {
+            string source = @"
+class O
+{
+    class N
+    {
+    }
+}
+";
+            TestTypeDefinitionDocuments(new[] { source },
+                ("O", "1.cs"),
+                ("N", "1.cs"));
```

Comment:
>("N", "1.cs")); [](http://example.com/codeflow?start=16&length=15)

No need to add N since we can find it by getting documents of the outer class.